### DISCUSSION TITLE
ActivityPub: Connecting my website to mastodon

### DISCUSSION THREAD

**Discussion by AmarOk1412** [mastodon/mastodon#22735] [Q&A] [answered]
### Steps to reproduce the problem

Hi everyone and merry Xmas.

This issue is not a bug from mastodon, but more from my side.

I'm working on a website for publishing recipes and I want to connect it to the fediverse

I read http-signatures/webfinger/activitypub documentation and got a minimal server working. I can discover @[chef@cha-cu.it](mailto:chef@cha-cu.it) on the fediverse, so far so good

However, I'm now wondering why new articles doesn't pop in my mastodon view.

I tried by posting the article in my [mastodon.social](http://mastodon.social/) inbox and just by waiting for instances to read my outbox without success.

So 2 questions:

+ Do you know when mastodon refresh the profile and read the outbox? Or do I really need to POST in the inbox?
+ Do you know if my outbox is malformed? If I need to change something in the object (outbox is there: https://cha-cu.it/users/chef/outbox) If I check https://docs.joinmastodon.org/spec/activitypub/#properties-used i think I use all the things I need?

I have a local instance but for some reason, when following it got some timeout I don't have with external instances so it's not easy to debug without the logs :/

Thank you in advance.

### Expected behaviour

Ideally, my posts should be sent to mastodon

### Actual behaviour

For now, I don't see the posts.

### Specifications

Mastodon (master)

**Comment by nightpool** [comment]
> Do you know when mastodon refresh the profile and read the outbox? Or do I really need to POST in the inbox?

ActivityPub is a push based protocol designed for realtime messaging. All posts need to be POSTed to the following servers. This is a departure from protocols like RSS, which have lots of well known issues with latency and slow updates.

> However, I'm now wondering why new articles doesn't pop in my mastodon view.

Did you Follow the https://cha-cu.it/users/chef account from the mastodon server in question? did you Accept the follow correctly, and does it show as accepted in the mastodon.social interface? If you just POST the activity unsolicited, then mastodon.social will discard the activity because it doesn't know what to do with it, you need to make sure that you have a follower on the other side that's expecting the notification

**Reply by AmarOk1412** [reply]
For the POST, make sense.

And yes, I follow @chef@cha-cu.it on mastodon, and the Accept is POSTED and seems accepted (in mastodon.social, I see "UNFOLLOW" and got it in my following list and I don't have any error returned (it was the case at the beginning due to the http-signatures library I used, but it's fixed)).

On a refresh, I posted (to https://mastodon.social/users/AmarOk/inbox):

```json
{
      "actor": "https://cha-cu.it/users/chef",
      "cc": [
        "https://cha-cu.it/users/chef/followers"
      ],
      "id": "https://cha-cu.it/recettes/crepes",
      "object": {
        "attributedTo": "https://cha-cu.it/recettes",
        "cc": [
          "https://cha-cu.it/users/chef/followers"
        ],
        "content": "---\ntitle: Crêpes\ndate: 2022-11-20\ntags: [\"sucré\", \"repas\"]\nduration: 1h\n---\n\n# Ingrédients\n\n+ 300g farine\n+ 75cl lait\n+ 3c à soupe d'huile\n+ 3c à soupe de sucre\n+ 3 œufs\n+ Bonus: rhum ou bière.\n\n# Équipement\n\n+ 1 Bol\n+ 1 Fouet\n+ 1 louche\n+ 1 poele\n\n||\n# Préparation\n\n1. Battre les œufs, ajouter le lait, l'huile et ajouter peu à peu le sucre puis la farine en mélangeant\npour obtenir un liquide homogène.\n2. Chauffer la poêle à température moyenne\n3. Ajouter un peu de beurre à la poële\n4. Cuire les crêpes une par une. On peut la retourner quand le contour commence à brunir.\n\n# Notes\n\n1. Se garnit très bien avec de la saumon fumé/sauce tartare/ciboulette ou jambon/emmental/œuf mirroir en\nrepas\n2. Sinon du miel, pâte à tartiner ou sirop d'érable ambré.\n3. Une version végan se fait très bien en remplacant le lait par du lait d'amandes et les œufs par de la\nfécule de maïs ou de pomme de terre.",
        "id": "https://cha-cu.it/recettes/crepes",
        "license": "CC-0",
        "mediaType": "text/markdown",
        "name": "Crêpes",
        "published": "2022-12-03T21:01:38.665111508+00:00",
        "to": [
          "https://www.w3.org/ns/activitystreams#Public"
        ],
        "type": "Article",
        "url": "https://cha-cu.it/recettes/crepes"
      },
      "published": "2022-12-03T21:01:38.665111508+00:00",
      "to": [
        "https://www.w3.org/ns/activitystreams#Public"
      ],
      "type": "Create"
    }
```

With a http-signature mechanism I use for the "Accept" activity (which is ok).

I post to the inbox (not the ["endpoints"]["sharedInbox"] one) and I don't have an error returned, so I think it's ok. I think something in the object may be wrong, but I don't see what can it be. "published" seems ok. Not sure if markdown is accepted.

**Reply by AmarOk1412** [reply]
Finally, I guess sensitive:false or attachments were missing. Now it works.

**Comment by AmarOk1412** [comment, marked as answer]
Finally, I guess sensitive:false or attachments were missing. Now it works.
